builtin/providers/google/data_source_google_compute_network_test.go

```diff
@@ -0,0 +1,74 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"testing"
+)
+
+func TestAccDataSourceGoogleNetwork(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: TestAccDataSourceGoogleNetworkConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceGoogleNetworktCheck("data.google_compute_network.my_network"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGoogleNetworktCheck(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("root module has no resource called %s", name)
+		}
+
+		networkOrigin, ok := s.RootModule().Resources["google_compute_network.foobar"]
```

Comment:
A few ideas to make this function more reusable (in case more tests come for this down the road):
- google_compute_network.foobar could be a parameter
- All attributes could be compared resource vs data source (right now id and self_link are doing that, but name and description are using hardcoded values)
Sound reasonable?